azure-kusto-data/azure/kusto/data/request.py

```diff
@@ -324,14 +329,81 @@ def _execute(self, endpoint, database, query, default_timeout, properties=None):
 
         raise KustoServiceError([response.json()], response)
 
+    def execute_streaming_ingest(
+        self,
+        database,
+        table,
+        stream,
+        stream_format,
+        mapping_name=None,
+        accept=None,
+        accept_encoding=None,
+        connection=None,
+        content_length=None,
+        content_encoding=None,
+        expect=None,
+    ):
+        """Executes streaming ingest against this client.
+        :param str database: Target database.
+        :param str table: Target table.
+        :param stream: File-like object contains the data to ingest.
+        :param DataFormat stream_format: Format of the data in the stream.
+        :param str mapping_name: Pre-defined mapping of the table. Required when stream_format is json/avro.
+        Other optional params: optional request headers as documented at:
+            https://kusto.azurewebsites.net/docs/api/rest/streaming-ingest.html
+        """
+
+        assert isinstance(database, str), "Expected str instance, found {}".format(type(database))
+        assert isinstance(table, str), "Expected str instance, found {}".format(type(table))
+        assert isinstance(stream_format, str), "Expected str instance, found {}".format(type(stream_format))
+
+        request_params = {"streamFormat": stream_format}
+
+        if stream_format == "json" or stream_format == "singlejson" or stream_format == "avro":
+            request_params["mappingName"] = mapping_name
+
+        headers = {
+            "Accept": accept,
+            "Accept-Encoding": accept_encoding or "gzip,deflate",
+            "Connection": connection or "Keep-Alive",
+            "Content-Length": str(content_length),
+            "Content-Encoding": content_encoding,
+            "Expect": expect,
+            "x-ms-client-version": "Kusto.Python.Client:" + VERSION,
+            "x-ms-client-request-id": "KPC.execute;" + str(uuid.uuid4()),
+        }
+
+        # Remove headers that are None
+        request_headers = {k: v for k, v in headers.items() if v is not None}
+
+        if self._auth_provider:
+            request_headers["Authorization"] = self._auth_provider.acquire_authorization_header()
+
+        request_headers["Host"] = self._streaming_ingest_endpoint.split("/")[2]
+
+        request_payload = stream

```

Comment:
we don't really need these dangling variables, would just put the value int he actual call .. 

same for `endpoint` and `timeout`